distributed/cli/dask_worker.py

```diff
@@ -167,9 +186,27 @@ def f():
                 json.dump(msg, f)
         loop.add_callback(f)
 
+    retired = [False]
+
+    if lifetime:
+        @gen.coroutine
+        def retire():
+            yield gen.sleep(lifetime)
+            logger.info("Retiring worker...")
+            with rpc(ip=nannies[0].scheduler.ip,
+                     port=nannies[0].scheduler.port) as scheduler:
+                workers = ([n.worker_address for n in nannies
+                            if n.process and n.worker_port] if nanny else
+                           [n.address for n in nannies])
+                scheduler.retire_workers(workers=workers, remove=False)
+            retired[0] = True
+            logger.info("Worker retired")
+
+        loop.add_callback(retire)
```

Comment:
Yes that makes sense regarding tests - I will have a go at moving it in when I get a chance. The only complication that comes to mind is that to survive restarts etc. the time might have to be converted to an absolute time.